sql/rpl_binlog_sender.cc

```diff
@@ -1030,7 +1038,73 @@ int Binlog_sender::send_format_description_event(IO_CACHE *log_cache,
   if (event_checksum_on() && event_updated)
     calc_event_checksum(event_ptr, event_len);
 
-  DBUG_RETURN(send_packet());
+  if (m_event_checksum_alg != binary_log::BINLOG_CHECKSUM_ALG_UNDEF &&
+      m_event_checksum_alg != binary_log::BINLOG_CHECKSUM_ALG_OFF)
+    event_len-= BINLOG_CHECKSUM_LEN;
+
+  Format_description_log_event *new_fdle= NULL;
+
+  try 
+  {
+    new_fdle= new Format_description_log_event(reinterpret_cast<char*>(event_ptr), event_len, m_fdle.get());
+  }
+  catch(...)
```

Comment:
The previous PR question still stands

Reply:
I have looked over the code and I cannot really prove that it can throw an exception. I saw quite a number of crashes when I was sending garbage data to slave, so just to be sure I put catch there. I am willing to remove it and just put no throwing new there.

Reply:
Please do, (and please check if this is not compiled with -fno-exceptions in the first place so that nothrow new would be also redundant)

Reply:
nothrow is redundant because new operator of class Log_event is overloaded and uses my_malloc to allocate memory.